src/codewhisperer/service/recommendationHandler.ts

```diff
@@ -270,6 +273,18 @@ export class RecommendationHandler {
                 errorCode = error.code
                 reason = `CodeWhisperer Invocation Exception: ${error?.code ?? error?.name ?? 'unknown'}`
                 await this.onThrottlingException(error, triggerType)
+
+                if (error?.code === 'AccessDeniedException') {
+                    getLogger().error('CodeWhisperer AccessDeniedException : %s', (error as Error).message)
+                    vscode.window
+                        .showErrorMessage(`CodeWhisperer: ${error?.message}`, CodeWhispererConstants.settingsLearnMore)
+                        .then(async resp => {
+                            if (resp === CodeWhispererConstants.settingsLearnMore) {
+                                openUrl(vscode.Uri.parse(CodeWhispererConstants.learnMoreUri))
+                            }
+                        })
+                    await vscode.commands.executeCommand('aws.codeWhisperer.disableAutoTriggerOfCodeSuggestions')
```

Comment:
if this is only called programmatically why isn't it just a function call? why was it defined as a vscode command?

Reply:
it is setting and calling other commands and state similar to the pattern of the `enableCodeSuggestions` command. We were just trying to keep similar process with similar.

Reply:
In fact it's almost identical to `enableCodeSuggestions`. So a parameter can be added to `enableCodeSuggestions` instead of defining this new command.